### PR TITLE
Update identity docs

### PR DESCRIPTION
**Pull request by rnicoll**
Update identity docs for open source, and merge in material from the wiki with a view to the open source docs replacing the wiki as the authoritative source of this information.

### CODE REVIEW

**Comment by rnicoll** [comment]
@joeldudleyr3 I've now removed myself from the Corda organisation, can you find someone to take this over please :)

File: docs/source/permissioning-node-name.rst [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    * No leading or trailing whitespace
+    * Does not include the following characters: ``,`` , ``=`` , ``$`` , ``"`` , ``'`` , ``\``
+    * Is in NFKC normalization form
+    * Does not contain the null character
+    * Only the latin, common and inherited unicode scripts are supported
+
+* The organisation field of the name also obeys the following constraints:
+
+    * No double-spacing
+    * Does not contain the words "node" or "server"
```
Comment (gendal) [inline docs/source/permissioning-node-name.rst]: i know there's been some debate about this but can we at least explain in the docs why this is.  (Also - is this case-sensitive? do we have a battery of tests?)

**Comment by joeldudleyr3** [comment]
@gendal I'll take this over. Is there any context I should have regarding these changes?

**Comment by joeldudleyr3** [comment]
Handing off to the network/doorman team. Required changes:
- Move information about certificates that nodes generate automatically to the "Corda nodes" section, and remove it from the "Permissioning" page
- Re-write to change from structure with "intermediate CA" to one with the Doorman CA and network map CA (ask Shams to find out more)
- General check for correctness following network/doorman team changes

File: docs/source/generating-a-node.rst [hunk 2]
```diff
@@ -34,6 +34,16 @@ set for X.509 certificates (specified in RFC 3280), plus the locality attribute:
 * Organizational-unit (OU)
 * Common name (CN) (only used for service identities)
```
Comment (kasiastreich) [inline docs/source/generating-a-node.rst]: No more service identities in CN.

**Comment by mikehearn** [comment]
@kasiastreich Who owns this PR now? Is it you?

**Comment by kasiastreich** [comment]
Good question, I can take over.

**Comment by mikehearn** [comment]
Super duper

File: docs/source/permissioning.rst [hunk 3]
```diff
@@ -34,6 +30,23 @@ A Corda network has three types of certificate authorities (CAs):
...
+* Confidential legal identity certificates are issued by a certificate marked as well known legal identity
+* Party certificates are marked as either a well known identity or a confidential identity
+* The structure of certificates above Doorman/Network map is intentionally left untouched, as they are not relevant to
+  the identity service and therefore there is no advantage in enforcing a specific structure on those certificates. The
+  certificate hierarchy consistency checks are required because nodes can issue their own certificates and can set
+  their own role flags on certificates, and it's important to verify that these are set consistently with the
+  certificate hierarchy design. As as side-effect this also acts as a secondary depth restriction on issued
+  certificates
+
+All the certificates except the Doorman certificates must be issued with the custom role extension (see below).
```
Comment (shamsasari) [inline docs/source/permissioning.rst]: Is this correct? Isn't it all the certificates above the doorman and network map certs don't have roles?

File: docs/source/permissioning.rst [hunk 4]
```diff
@@ -34,6 +30,23 @@ A Corda network has three types of certificate authorities (CAs):
...
+
 We can visualise the permissioning structure as follows:
 
 .. image:: resources/certificate_structure.png
```
Comment (shamsasari) [inline docs/source/permissioning.rst]: Add a TODO that this needs to be updated. I believe @kchalkias has the originals.

File: docs/source/permissioning.rst [hunk 5]
```diff
@@ -58,6 +71,27 @@ public/private keypairs and certificates. The keypairs and certificates should o
...
+
+In a typical installation, node administrators needn't be aware of these. However, when node certificates are managed
+by external tools (such as an existing PKI solution deployed within an organisation), it is important to understand
+these constraints.
+
+Certificate path validation is extended so that a certificate must contain the extension if the extension was present
+in the certificate of the issuer.
+
 Creating the root and intermediate network CAs
 ----------------------------------------------
```
Comment (shamsasari) [inline docs/source/permissioning.rst]: For the bullet-point 4 below, include the trust store needs to be given to the node before it does registration with the doorman.

Comment (shamsasari) [inline docs/source/permissioning.rst]: The reference to "intermediate" should change to "doorman"

File: docs/source/permissioning.rst (hunk 3 above)
Comment (kasiastreich) [reply docs/source/permissioning.rst]: I don't know, who should I ask about this bit of code? @mkit ?

Comment (mkit) [reply docs/source/permissioning.rst]: Yes @shamsasari  is correct, only the ROOT certificate has the certificate role set to null. All the rest have this extension set.
As such, I would rephrase this sentence to something like:
"All the certificates must be issued the custom role extension (see below)."
We are not mentioning the ROOT cert in this doc at all so it should be fine.

Comment (kchalkias) [reply docs/source/permissioning.rst]: @shamsasari was correct on his initial comment, it should say "All certificates except the *Root* certificate...". That said, I agree with @mkit's recommendation to omit a reference to the Root cert atm. Iirc, the reason for not setting a role to ROOT was to allow for more intermediate certs in the future.

- Although not related to this PR, we should note that this dynamic path-length is contradictory to our logic of checking if a certificate belongs to the NetworkMap, because in the `verifiedNetworkMapCert` we check if Root is its parent.

Comment (shamsasari) [reply docs/source/permissioning.rst]: It's still backwards compatible to change `verifiedNetworkMapCert` if we introduce any intermediate certs, right?

File: docs/source/generating-a-node.rst [hunk 6]
```diff
@@ -21,18 +21,32 @@ Each Corda node has the following structure:
 The node is configured by editing its ``node.conf`` file. You install CorDapps on the node by dropping the CorDapp JARs
 into the ``cordapps`` folder.
 
+In development mode (i.e. when ``devMode = true``, see :doc:`corda-configuration-file` for more information), the ``certificates``
+direcotry is filled in with pre-configured keystores if the required keystores do not exist. This ensures that developers
```
Comment (joeldudleyr3) [inline docs/source/generating-a-node.rst]: Typo ("direcotry").

"filled in" -> "filled"

File: docs/source/generating-a-node.rst [hunk 7]
```diff
@@ -21,18 +21,32 @@ Each Corda node has the following structure:
...
 
 * Organization (O)
 * State (ST)
 * Locality (L)
 * Country (C)
 * Organizational-unit (OU)
-* Common name (CN) (only used for service identities)
+* Common name (CN)
+
+Note that serial number is intentionally excluded in order to minimise scope for uncertainty in distinguished name format.
```
Comment (joeldudleyr3) [inline docs/source/generating-a-node.rst]: "the serial number", "the distinguished name format"

File: docs/source/generating-a-node.rst [hunk 8]
```diff
@@ -21,18 +21,32 @@ Each Corda node has the following structure:
...
+Distinguished name qualifier has been removed due to technical issues; consideration was given to "Corda" as qualifier,
```
Comment (joeldudleyr3) [inline docs/source/generating-a-node.rst]: "The distinguished name qualifier"

File: docs/source/generating-a-node.rst [hunk 9]
```diff
@@ -21,18 +21,32 @@ Each Corda node has the following structure:
...
+however the qualifier needs to reflect directory manager, not the technology involved. There may be many Corda directories,
```
Comment (joeldudleyr3) [inline docs/source/generating-a-node.rst]: "the directory manager".

What is meant by "directory" here?

File: docs/source/generating-a-node.rst [hunk 10]
```diff
@@ -21,18 +21,32 @@ Each Corda node has the following structure:
...
+but only one R3 directory on Corda. The ordering of attributes is important.
+
+State should be avoided unless required to differentiate from other localities with the same or similar names at the
```
Comment (joeldudleyr3) [inline docs/source/generating-a-node.rst]: We should put references to these name attributes in double-backticks to make it clear we are referring to the property of the name.

File: docs/source/generating-a-node.rst [hunk 11]
```diff
@@ -21,18 +21,32 @@ Each Corda node has the following structure:
...
+country level. For example, London (GB) would not need a state, but St Ives would (there are two, one in Cornwall, one
+in Cambridgeshire). As legal entities in Corda are likely to be located in major cities, this is not expected to be
```
Comment (joeldudleyr3) [inline docs/source/generating-a-node.rst]: "this attribute is not expected"

File: docs/source/generating-a-node.rst [hunk 12]
```diff
@@ -67,6 +81,13 @@ The name must also obey the following constraints:
         * This is to avoid right-to-left issues, debugging issues when we can't pronounce names over the phone, and
           character confusability attacks
 
+External identifiers
+^^^^^^^^^^^^^^^^^^^^
+For mappings to external identifiers such as Companies House nos., LEI, BIC, etc. these should be stored in custom X.509
```
Comment (joeldudleyr3) [inline docs/source/generating-a-node.rst]: Change to "Mappings to external identifiers such as Companies House nos., LEI, BIC, etc. should be stored"

File: docs/source/generating-a-node.rst [hunk 13]
```diff
@@ -67,6 +81,13 @@ The name must also obey the following constraints:
...
+certificate extensions. These values may change due to operational reasons, without the identity they're associated with
```
Comment (joeldudleyr3) [inline docs/source/generating-a-node.rst]: "due to" -> "for"

File: docs/source/generating-a-node.rst [hunk 14]
```diff
@@ -67,6 +81,13 @@ The name must also obey the following constraints:
...
+necessarily changing, and their inclusion in the distinguished name would cause significant logistical complications.
+OID and format for these extensions will be described in a further specification.
```
Comment (joeldudleyr3) [inline docs/source/generating-a-node.rst]: "The OID and format"

File: docs/source/permissioning.rst [hunk 15]
```diff
@@ -10,30 +10,43 @@ Corda networks are *permissioned*. To connect to a network, a node needs three k
...
     key signing to reduce the risk of the root network CA's private key being compromised
 
 * The **node CAs**
 
   * Each node serves as its own CA in issuing the child certificates that it uses to sign its identity
     keys and TLS certificates
 
+The following constraints are also imposed:
+
+* Doorman certificates are issued by a certificate
```
Comment (joeldudleyr3) [inline docs/source/permissioning.rst]: Are issued by which certificate?

File: docs/source/permissioning.rst [hunk 16]
```diff
@@ -10,30 +10,43 @@ Corda networks are *permissioned*. To connect to a network, a node needs three k
...
+* Well-known service identity certificates are issued by a certificate marked as Doorman
```
Comment (joeldudleyr3) [inline docs/source/permissioning.rst]: Here and in the following lines, which is actually meant by "marked"?

File: docs/source/permissioning.rst [hunk 17]
```diff
@@ -74,31 +108,32 @@ Creating the root network CA's keystore and truststore
...
 
-   * This keystore will be used by the root network CA to sign the intermediate network CA's certificate
+   * This keystore will be used by the root network CA to sign the doorman CA's certificate
 
 4. Create a new Java keystore named ``truststore.jks`` and store the root network CA's certificate in it using the
    alias ``cordarootca``
 
-   * This keystore will be provisioned to the individual nodes later
+   * This keystore will be provisioned to the individual nodes later. The `truststore.jks` needs to be given to the node
+    before it does registration with the doorman.
```
Comment (joeldudleyr3) [inline docs/source/permissioning.rst]: "does registration" -> "registers"

File: docs/source/permissioning.rst [hunk 18]
```diff
@@ -74,31 +108,32 @@ Creating the root network CA's keystore and truststore
...
 3. Create a new keystore and store the root network CA's keypair and certificate in it for later use
 
-   * This keystore will be used by the root network CA to sign the intermediate network CA's certificate
+   * This keystore will be used by the root network CA to sign the doorman CA's certificate
 
 4. Create a new Java keystore named ``truststore.jks`` and store the root network CA's certificate in it using the
    alias ``cordarootca``
 
-   * This keystore will be provisioned to the individual nodes later
+   * This keystore will be provisioned to the individual nodes later. The `truststore.jks` needs to be given to the node
```
Comment (joeldudleyr3) [inline docs/source/permissioning.rst]: What is actually meant by "needs to be given"?

File: docs/source/permissioning.rst [hunk 19]
```diff
@@ -10,17 +10,13 @@ Corda networks are *permissioned*. To connect to a network, a node needs three k
...
+Production deployments require a secure certificate authority.
 Most production deployments will use an existing certificate authority or construct one using software that will be
 made available in the coming months. Until then, the documentation below can be used to create your own certificate
 authority.
 
-Network structure
------------------
+Certificate hierarchy
+---------------------
 A Corda network has three types of certificate authorities (CAs):
```
[the remark this answers is not in the document]
Comment (kchalkias) [reply docs/source/permissioning.rst]: @shamsasari what about this?

[the remark this answers is not in the document]
Comment (shamsasari) [reply docs/source/permissioning.rst]: You are correct. This list needs to include legal identity as well.

File: docs/source/release-notes.rst [hunk 20]
```diff
@@ -8,7 +8,7 @@ Unreleased
 * X.509 certificates now have an extension that specifies the Corda role the certificate is used for, and the role
   hierarchy is now enforced in the validation code. This only has impact on those developing integrations with external
   PKI solutions, in most cases it is managed transparently by Corda. A formal specification of the extension can be
-  found at see :doc:`permissioning-certificate-specification`.
+  found at see :doc:`permissioning`.
```
Comment (kchalkias) [inline docs/source/release-notes.rst]: remove `see`

File: docs/source/permissioning.rst (hunk 19 above)
[the remark this answers is not in the document]
Comment (kchalkias) [reply docs/source/permissioning.rst]: @kasiastreich let's change `three` to `four` and add well-known identity in the list of CAs (apart from signing transactions, it issues the certs for confidential legal identities).

File: docs/source/permissioning.rst [hunk 21]
```diff
@@ -30,6 +30,10 @@ A Corda network has three types of certificate authorities (CAs):
   * Each node serves as its own CA in issuing the child certificates that it uses to sign its identity
     keys and TLS certificates
 
+* The **legal identity** CAs
+
+   * Node's legal identity apart from signing transactions can also issue certificates for confidential legal identities
```
Comment (kchalkias) [inline docs/source/permissioning.rst]: Node's _well-known legal identity_, apart from signing transactions, it can ...

File: docs/source/permissioning.rst [hunk 22]
```diff
@@ -30,6 +30,10 @@ A Corda network has three types of certificate authorities (CAs):
   * Each node serves as its own CA in issuing the child certificates that it uses to sign its identity
     keys and TLS certificates
 
+* The **legal identity** CAs
```
Comment (kchalkias) [inline docs/source/permissioning.rst]: **legal identity CAs**
